### PR TITLE
Add `supports_submit_qubo` filter to `backends()`

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Fixes an issue with `ss_unconstrained_simulator` being excluded from a `qiskit_superstaq.SuperstaqProvider().backends()` call after recent updates such as https://github.com/Infleqtion/client-superstaq/pull/914.

### CODE REVIEW

**Comment by bharat-thotakura** [comment]
(The diff in the notebook is due to a ruff format automatically applied locally, the main change in it is the mention of the `supports_submit_qubo` filter -- see https://github.com/Infleqtion/client-superstaq/pull/922#issuecomment-1986011841 for visualization)

**Comment by richrines1** [comment]
oh wow good catch, this is my bad :)

File: docs/source/get_started/access_info/access_info_qss.ipynb
Comment (richrines1) [inline docs/source/get_started/access_info/access_info_qss.ipynb:1]: it looks like the formatter is indenting the notebook's json more than jupyter does? in which case it's probably good to undo before we merge (otherwise it will probably just be reverted in the next commit which touches this file and we'll end up with a bunch of unnecessary changes in our git history)

Comment (bharat-thotakura) [reply docs/source/get_started/access_info/access_info_qss.ipynb:1]: Sure, will revert
